src/download.cpp

```diff
@@ -206,6 +206,13 @@ void DownloadFile(const std::string& url, IDownloadSink *sink, int flags)
 
     if ( !filename_set )
     {
+        char ioptionurl[512];
```

Comment:
I used a different buffer to make sure there is a backup url to use in case InternetQueryOptionA fails somehow and ends up emptying the original buffer, leaving the app without a url to try at all. Surely this should never happen, but just covering all bases.
In any case either works for me, I've already changed it to use a dynamic size buffer, but I'm happy to change it to whatever you prefer.